### PR TITLE
[Tokentube] Add extractor

### LINKED ISSUE #800: [Site Request] TokenTube
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running yt-dlp version **2021.08.10**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] The provided URLs do not contain any DRM to the best of my knowledge
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://tokentube.net/l/3236632011/Praise-A-Thon-Pastori-Chrisin-ja-Pastori-Bennyn-kanssa-27-8-2021
- Channel: https://tokentube.net/channel/3697658904/TokenTube

## Description

Hi, can you add support for TokenTube? It's a Finnish alternative to YouTube.

### PR DESCRIPTION
**Pull request by u-spec-png**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [x] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

Adds `TokentubeIE` and `TokentubeChannelIE` closes #800.

### CODE REVIEW

File: yt_dlp/extractor/tokentube.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        title = self._html_search_regex(r'<h1\s*class=["\']title-text["\']>(.+?)</h1>', webpage, 'title')
+
+        data_json = self._html_search_regex(r'({["\']html5["\'].+?}}}+)', webpage, 'data json')
+        data_json = self._parse_json(js_to_json(data_json), video_id, fatal=False)
+
+        sources = data_json.get('sources')
+        if not sources:
+            sources = self._html_search_regex(r'updateSrc\(([^\)]+)\)', webpage, 'sources')
+            sources = self._parse_json(js_to_json(sources), video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/tokentube.py:67]:
```suggestion
            sources = data_json.get('sources') or self._parse_json(
                self._html_search_regex(r'updateSrc\(([^\)]+)\)', webpage, 'sources'),
                video_id, transform_source=js_to_json)
```

File: yt_dlp/extractor/tokentube.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        for format in sources:
+            formats.append({
+                'url': format.get('src'),
+                'format_id': format.get('label'),
+                'height': format.get('res'),
+            })
```
Comment (pukkandan) [inline yt_dlp/extractor/tokentube.py:74]:
```suggestion
        formats = [{
            'url': format.get('src'),
            'format_id': format.get('label'),
            'height': format.get('res'),
        } for format in sources]
```

**Review by pukkandan — COMMENTED** [review]
I haven't gone through the regexes yet, but everything else looks good

File: yt_dlp/extractor/tokentube.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+class TokentubeIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?tokentube\.net/(?:view)?/?\??[lv]=?/?(?P<id>\d+)(?:/.*)?'
+    _TESTS = [{
+        'url': 'https://tokentube.net/l/3236632011/Praise-A-Thon-Pastori-Chrisin-ja-Pastori-Bennyn-kanssa-27-8-2021',
+        'info_dict': {
+            'id': '3236632011',
+            'ext': 'mp4',
+            'title': 'Praise-A-Thon Pastori Chrisin ja Pastori Bennyn kanssa 27.8.2021',
+            'description': '',
```
Comment (pukkandan) [inline yt_dlp/extractor/tokentube.py:24]: Is the description actually empty, or we just cant find it?

Comment (u-spec-png) [reply yt_dlp/extractor/tokentube.py:24]: Description is empty.

**Pull request merged**